omegaconf/listconfig.py

```diff
@@ -252,12 +254,15 @@ def __setitem__(self, index: Union[int, slice], value: Any) -> None:
             self._format_and_raise(key=index, value=value, cause=e)
 
     def append(self, item: Any) -> None:
+
         try:
             from omegaconf.omegaconf import OmegaConf, _maybe_wrap
 
             index = len(self)
             self._validate_set(key=index, value=item)
 
+            item = self._resolve_container_assignment(item)
+
             node = _maybe_wrap(
                 ref_type=self.__dict__["_metadata"].element_type,
                 key=index,
```

Comment:
see, here for example the responsibility of _resolve_container_assignment() can be split between validate_set() above and _maybe_wrap() below.